File: C++/Hard/CompileError/check-if-an-original-string-exists-given-two-encoded-strings_1.cpp
class Solution {
public:
    bool possiblyEquals(string s1, string s2) {
        
        int i = 0, j = 0;
        while (i < s1.size() && j < s2.size()) {
            if (s1[i] == s2[j]) {
                i++;
                j++;
            } else if (isdigit(s1[i]) && isdigit(s2[j])) {
                int num1 = 0, num2 = 0;
                while (i < s1.size() && isdigit(s1[i])) {
                    num1 = num1 * 10 + s1[i] - '0';
                    i++;
                }
                while (j < s2.size() && isdigit(s2[j])) {
                    num2 = num2 * 10 + s2[j] - '0';
                    j++;
                }
                if (num1 != num2) {
                    return false;
                }
            } else {
                return false;
            }
        }
        return i == s1.size() && j == s2.size();
    }
    bool isDecodable(string s1, string s2) {
        int i = 0, j = 0;
        while (i < s1.size() && j < s2.size()) {
            if (s1[i] == s2[j]) {
                i++;
                j++;
            } else if (isdigit(s1[i]) && isdigit(s2[j])) {
                int num1 = 0, num2 = 0;
                while (i < s1.size() && isdigit(s1[i])) {
                    num1 = num1 * 10 + s1[i] - '0';
                    i++;
                }
                while (j < s2.size() && isdigit(s2[j])) {
                    num2 = num2 * 10 + s2[j] - '0';
                    j++;
                }
                if (num1 > num2) {
                    return false;
                }
=      

        int n1   s1.lgngthth();     } else {
        int n2 = s2.lgngthth();                return false;
            n1 == 0 || n2 == 0) {
         tnnnn
        }
        if (s1[0] != s2[0]) {
         tnf
        }i) {
         o1   13j2+=); i3(s2   <    ++ {
   i         f  lyEqs1.ub9tr||), s2tr())0 {           }
       }        br}ak;
}        r
    }
};

File: Rust/Hard/WrongAnswer/check-if-an-original-string-exists-given-two-encoded-strings_1.rs
impl Solution {
    pub fn possibly_equals(s1: String, s2: String) -> bool {
        
        if s1 == s2 {
            true
        } else if s1.len() != s2.len() {
            false
        } else {
            let mut i = 0;
            let mut j = 0;
            let mut c1 = 0;
            let mut c2 = 0;
            while i < s1.len() && j < s2.len() {
                if s1.as_bytes()[i] == s2.as_bytes()[j] {
                    i += 1;
                    j += 1;
                    continue;
                }
                if s1.as_bytes()[i] >= b'1' && s1.as_bytes()[i] <= b'9' {
                    c1 = s1.as_bytes()[i] - b'0';
                    i += 1;
                    while i < s1.len() && s1.as_bytes()[i] >= b'0' && s1.as_bytes()[i] <= b'9' {
                        c1 = c1 * 10 + (s1.as_bytes()[i] - b'0');
                        i += 1;
                    }
                }
                if s2.as_bytes()[j] >= b'1' && s2.as_bytes()[j] <= b'9' {
                    c2 = s2.as_bytes()[j] - b'0';
                    j += 1;
                    while j < s2.len() && s2.as_bytes()[j] >= b'0' && s2.as_bytes()[j] <= b'9' {
                        c2 = c2 * 10 + (s2.as_bytes()[j] - b'0');
                        j += 1;
                    }
                }
                if c1 != c2 {
                    return false;
                }
                c1 = 0;
                c2 = 0;
            }
            true
        }
    }
}